src/helper/shallowEqualSelector.js

```diff
@@ -9,9 +9,8 @@
  * @providesModule shallowEqualSelector
  */
 
-'use strict'
+'use strict';
 
-import { createSelectorCreator, defaultMemoize } from 'reselect';
-import shallowEqual from 'shallowEqual';
+import { createSelector } from '@reduxjs/toolkit';
 
-export default createSelectorCreator(defaultMemoize, shallowEqual);
+export default createSelector;
```

Comment:
This selector is used inside reducer. redux-toolkit wraps the states in a proxy to detect mutations internally. When we use this selector inside a reducer, it tries to compare the prev state and current state. Prev state will be a revoked proxy because scope of prev reducer call has ended. When we try to perform shallw comparision on prev state(revoked proxy) and current state, we do use some methods like Object.keys() on states which results in an error because prev state is a revoked proxy. So Need to remove the custom shakkow check of prev and current states.

Reply:
Wouldn't it be better to replace everywhere that uses shallowEqualSelector with createSelector directly?

Also have you done any analysis of performance issues which might be introduced by doing this?  Are we helped by redux-toolkit introducing Immer, or is there a cost here we may want to address?

Reply:
It can be used outside of reducer to do a shallow comparision as state won't be a revoked proxy there.

Reply:
Regarding performance, redux-toolkit is almost as fast as native reducers except when we have a lot of data (approx more than 1k objects) because immer freezez the state after it is returned from slice. So, immer has to go deep down to every level of a nested object and freeze everything resulting in a little bit performance impact. We can deal with it by freezing the 1st layer manually before returning to make immer skip the freezing process.
One more advantage of using immer is, it prevents us mutating state inside a component as it has freezed state.

Reply:
Sorry got things cleared up.
Let's say we have 10k objects and we mutate a single aboject among those, then immer will freeze only single object and other 9999 objects wll remain same. Freezing wiill impact performance only if that 1 object is deepl nested and we change data on multiple levels of the object.

Reply:
https://immerjs.github.io/immer/docs/introduction
It says that "immer structurally sharing the things that weren't changed".
So, if we only mutatea single object out of 10k objects, when it try to freeze them again, it will only freeze single object and will not touch other 9999 objects

Reply:
I say we don't bother about freezing at all since it's pretty standard to treat store state data as immutable. 
I also don't want Immer to bother about freezing.
From the [docs](https://immerjs.github.io/immer/docs/freezing), turning off freezing should be as simple as running:
```js
import { setAutoFreeze } from 'immer'; 
setAutoFreeze(false);
```

For dev/debugging purposes, if we wanna have the check, we can have `setAutoFreeze(__DEV__);`